### PR TITLE
Add secret-type to Secret Management Core

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
### Proposed changes in this pull request

> This is a improvement to [#3630](https://github.com/wso2/carbon-identity-framework/pull/3630)
> Added a secret type so that a Secret belongs to a secret type and a tenant domain.
> Related issue - [#12264](https://github.com/wso2/product-is/issues/12264)

### CODE REVIEW

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 1]
```diff
@@ -98,13 +104,15 @@ public Secret getSecret(String secretName) throws SecretManagementException {
...
 
     @Override
-    public Secrets getSecrets() throws SecretManagementException {
+    public Secrets getSecrets(String secretTypeName) throws SecretManagementException {
 
         validateSecretManagerEnabled();
-        List secretList = this.getSecretDAO().getSecrets(getTenantId());
+        validateSecretsRetrieveRequest(secretTypeName);
+        SecretType secretType = getSecretType(secretTypeName);
+        List secretList = this.getSecretDAO().getSecrets(secretType.getId(), secretType.getName(), getTenantId());
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:112]: why we need to pass secretType and name ?
If we want both, better to pass the SecretType object itself

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java [hunk 2]
```diff
@@ -67,7 +69,7 @@
      * @param tenantId Id of the tenant.
      * @return A list of {@link Secret} for the tenant
      */
-    List getSecrets(int tenantId) throws SecretManagementException;
+    List getSecrets(String secretTypeId, String secretTypeName, int tenantId) throws SecretManagementException;
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java:72]: Why do we need secretType ID and Name ?

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java [hunk 3]
```diff
@@ -85,7 +87,7 @@
      * @param tenantId Tenant id of the {@link Secret}.
      * @throws SecretManagementException Secret Management Exception.
      */
-    void deleteSecretByName(String name, int tenantId) throws SecretManagementException;
+    void deleteSecretByName(String name, String secretTypeId, int tenantId) throws SecretManagementException;
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java:90]: Why do we need secretType ID and Name ?

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 4]
```diff
@@ -289,7 +380,7 @@ private SecretDAO getSecretDAO() throws SecretManagementException {
     private void validateSecretManagerEnabled() throws SecretManagementServerException {
 
         if (!SecretManagerComponentDataHolder.getInstance().isSecretManagementEnabled()) {
-            throw handleServerException(ERROR_CODE_SECRET_MANAGER_NOT_ENABLED, DB_TABLE_NAME);
+            throw handleServerException(ERROR_CODE_SECRET_MANAGER_NOT_ENABLED, DB_TABLE_SECRET);
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:383]: let's not throw the table name to the caller. This should be internal detail.

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 5]
```diff
@@ -361,4 +452,119 @@ private String encrypt(String plainText) throws CryptoException {
...
+
+        if (!isSecretTypeExists(secretTypeName)) {
+            if (log.isDebugEnabled()) {
+                log.debug("A secret type with the name: " + secretTypeName + " does not exists.");
+            }
+            throw handleClientException(ERROR_CODE_SECRET_TYPE_DOES_NOT_EXISTS, secretTypeName);
+        }
+    }
+
+    private String generateSecretTypeId(String secretTypeName) throws SecretManagementException {
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:536]:
```suggestion
    private String RetriveOrGenerateSecretTypeId(String secretTypeName) throws SecretManagementException {
```

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 6]
```diff
@@ -327,11 +418,11 @@ private boolean isSecretExistsById(String secretId) throws SecretManagementExcep
         return this.getSecretDAO().isExistingSecret(secretId, getTenantId());
     }
 
-    private String generateSecretId(String secretName) throws SecretManagementException {
+    private String generateSecretId(String secretTypeName, String secretName) throws SecretManagementException {
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:421]:
```suggestion
    private String RetriveOrGenerateSecretId(String secretTypeName, String secretName) throws SecretManagementException {
```

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 7]
```diff
@@ -224,18 +315,18 @@ private void validateSecretDeleteRequest(String secretName)
...
+    private void validateSecretCreateRequest(String secretTypeName, Secret secret) throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
             throw handleClientException(ERROR_CODE_SECRET_ADD_REQUEST_INVALID, null);
         }
-        if (isSecretExist(secret.getSecretName())) {
+        if (isSecretExist(secretTypeName, secret.getSecretName())) {
             if (log.isDebugEnabled()) {
                 log.debug("A secret with the name: " + secret.getSecretName() + " does not exists.");
```
Comment (senthalan) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:331]:
```suggestion
                log.debug("A secret with the name: " + secret.getSecretName() + " does exists.");
```

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:112]: This is name of the secret Type and name(alias) of the secret.
eg- secret type: adaptive
      secretname : choreoSampleService1

Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java]: resolved

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java (hunk 2 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java]: resolved

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java (hunk 3 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java:90]: resolved

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 4 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java]: resolved

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 5 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java]: resolved

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 6 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java]: resolved

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 7 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java]: resolved

**Review by senthalan — APPROVED** [review]

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java [hunk 8]
```diff
@@ -30,36 +31,40 @@
...
      */
-    Secrets getSecrets() throws SecretManagementException;
+    Secrets getSecrets(String secretTypeName) throws SecretManagementException;
 
     /**
      * This API is used to delete the given secret.
      *
-     * @param secretName Request to delete the {@link Secret}.
+     * @param secretTypeName Name of the {@link SecretType}.
+     * @param secretName     Request to delete the {@link Secret}.
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java:64]: Shouldn't this be `Name of the {@link Secret}.`

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java [hunk 9]
```diff
@@ -81,8 +86,42 @@
     /**
      * This function is used to replace a given secret.
      *
-     * @param secret secret object.
+     * @param secretTypeName Name of the {@link SecretType}.
+     * @param secret         secret object.
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java:90]: Fix formatting. -->`The Secret object.`

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 10]
```diff
@@ -98,13 +104,15 @@ public Secret getSecret(String secretName) throws SecretManagementException {
...
 
         validateSecretManagerEnabled();
-        List secretList = this.getSecretDAO().getSecrets(getTenantId());
+        validateSecretsRetrieveRequest(secretTypeName);
+        SecretType secretType = getSecretType(secretTypeName);
+        List secretList = this.getSecretDAO().getSecrets(secretType, getTenantId());
         if (secretList == null) {
             if (log.isDebugEnabled()) {
-                log.debug("No secret found for the tenant: " + getTenantDomain());
+                log.debug("No secret found the secretTypeName: " + secretTypeName + "for the tenant: " + getTenantDomain());
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:115]: `log.debug("No secret found for the secretTypeName: " + secretTypeName + "for the tenant: " + getTenantDomain());`

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 11]
```diff
@@ -182,40 +192,121 @@ public Secret replaceSecret(Secret secret) throws SecretManagementException {
...
+        }
+    }
+
     /**
-     * Validate that secret name is non-empty.
+     * Validate that secret type and secret name is non-empty.
      *
+     * @param secretTypeName Name of the {@link SecretType}.
      * @param secretName The secret name.
      * @throws SecretManagementException If secret validation fails.
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:266]: Formatting issue

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 12]
```diff
@@ -182,40 +192,121 @@ public Secret replaceSecret(Secret secret) throws SecretManagementException {
...
     }
 
     /**
-     * Validate that secret name is non-empty.
+     * Validate that secret type and secret name is non-empty.
      * Set tenant domain if they are not set to the secret object.
      *
+     * @param secretTypeName Name of the {@link SecretType}.
      * @param secretName The secret name.
      * @throws SecretManagementException If secret validation fails.
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:296]: Formatting issue

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 13]
```diff
@@ -182,40 +192,121 @@ public Secret replaceSecret(Secret secret) throws SecretManagementException {
...
      */
-    private void validateSecretDeleteRequest(String secretName)
+    private void validateSecretDeleteRequest(String secretTypeName, String secretName)
             throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secretName)) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secretName)) {
             if (log.isDebugEnabled()) {
-                log.debug("Error identifying the secret with secret name: " + secretName + ".");
+                log.debug("Error identifying the secret with secret name: " + secretName + " and secret type:"
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:303]: `log.debug("Error identifying the secret with secret name: " + secretName + " and secret type: "`

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 14]
```diff
@@ -224,20 +315,20 @@ private void validateSecretDeleteRequest(String secretName)
...
      * Set tenant domain if they are not set to the secret object.
      *
      * @param secret The secret to be added.
      * @throws SecretManagementException If secret validation fails.
      */
-    private void validateSecretCreateRequest(Secret secret) throws SecretManagementException {
+    private void validateSecretCreateRequest(String secretTypeName, Secret secret) throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:326]: Seems like this has exceeded 120 character limit

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java [hunk 15]
```diff
@@ -247,20 +338,20 @@ private void validateSecretCreateRequest(Secret secret) throws SecretManagementE
...
      *
      * @param secret The secret to be replaced.
      * @throws SecretManagementException If secret validation fails.
      */
-    private void validateSecretReplaceRequest(Secret secret)
+    private void validateSecretReplaceRequest(String secretTypeName, Secret secret)
             throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:350]: Seems like this has exceeded 120 character limit

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java [hunk 16]
```diff
@@ -105,4 +107,36 @@
...
+     * @return {@link SecretType} for the given name.
+     * @throws SecretManagementException Secret Management Exception.
+     */
+    SecretType getSecretTypeByName(String secretTypeName) throws SecretManagementException;
+
+    /**
+     * Delete {@link SecretType} by name.
+     *
+     * @param secretTypeName Name of the {@link SecretType}.
+     * @throws SecretManagementException secret Management Exception.
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java:139]: `* @throws SecretManagementException Secret Management Exception.`

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/CachedBackedSecretDAO.java [hunk 17]
```diff
@@ -146,6 +143,31 @@ public boolean isExistingSecret(String secretId, int tenantId) throws SecretMana
...
+    public void addSecretType(SecretType secretType) throws SecretManagementException {
+
+        secretDAO.addSecretType(secretType);
+    }
+
+    @Override
+    public void replaceSecretType(SecretType secretType) throws SecretManagementException {
+
+        secretDAO.replaceSecretType(secretType);
+
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/CachedBackedSecretDAO.java:156]: Remove redundant line

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/SecretDAOImpl.java [hunk 18]
```diff
@@ -310,15 +403,18 @@ public void replaceSecret(Secret secret) throws SecretManagementException {
...
             template.executeUpdate(UPDATE_SECRET, preparedStatement -> {
-                preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_NAME, secret.getSecretName());
-                preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_VALUE, secret.getSecretValue());
+                preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_SECRET_NAME, secret.getSecretName());
+                preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_SECRET_VALUE, secret.getSecretValue());
                 preparedStatement.setTimeStamp(DB_SCHEMA_COLUMN_NAME_LAST_MODIFIED, currentTime, calendar);
+                preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_TYPE, secretType.getId());
+                preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_DESCRIPTION, secret.getDescription());
                 preparedStatement.setString(DB_SCHEMA_COLUMN_NAME_ID, secret.getSecretId());
+
```
Comment (sachiniWettasinghe) [inline components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/SecretDAOImpl.java:417]: remove redundant line

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java (hunk 8 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java:64]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java (hunk 9 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java:90]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 10 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:115]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 11 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:266]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 12 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:296]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 14 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:326]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 15 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:350]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/CachedBackedSecretDAO.java (hunk 17 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/CachedBackedSecretDAO.java:156]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/SecretDAOImpl.java (hunk 18 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/impl/SecretDAOImpl.java:417]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java (hunk 16 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/dao/SecretDAO.java:139]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

File: components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java (hunk 13 above)
Comment (ChanikaRuchini) [reply components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java:303]: Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)

**Pull request merged**